src/WallpaperEngine/Assets/CDirectory.cpp

```
@@ -1,61 +1,94 @@
+            delete [] contents;
+            throw CAssetLoadException (filename, "Unexpected error when reading the file");
+        }
 
-    if (length != nullptr)
-        *length = size;
+        if (length != nullptr) {
+            *length = size;
+        }
 
-    return contents;
+        return contents;
```

Cppcheck (2.10-2)

[error] Resource leak

(resourceLeak)